File: AgoraChat-ObjectiveC/AgoraChat/Code/Controllers/Settings/AgoraAboutViewController.m
/************************************************************
 *  * Hyphenate
 * __________________
 * Copyright (C) 2016 Hyphenate Inc. All rights reserved.
 *
 * NOTICE: All information contained herein is, and remains
 * the property of Hyphenate Inc.
 */

#import "AgoraAboutViewController.h"
#import "ACDTitleDetailCell.h"
#import <UIKit/UIKit.h>
#import "ACDWebViewController.h"

@interface AgoraAboutViewController ()

@end

@implementation AgoraAboutViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.title = @"";
    [self setNavBar];
    self.tableView.separatorStyle = UITableViewCellSeparatorStyleNone;
    self.tableView.backgroundColor = UIColor.whiteColor;
}

- (void)setNavBar {
    UIButton *backButton = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 8, 15)];
    [backButton setImage:[UIImage imageNamed:@"black_goBack"] forState:UIControlStateNormal];
    [backButton addTarget:self action:@selector(back) forControlEvents:UIControlEventTouchUpInside];
    
    [backButton setTitle:@"About" forState:UIControlStateNormal];
    [backButton setTitleColor:TextLabelBlackColor forState:UIControlStateNormal];
    
    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithCustomView:backButton];
}

- (void)back {
    [self.navigationController popViewControllerAnimated:YES];
}

#pragma mark - Table view data source
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return 3;
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    ACDTitleDetailCell *cell = [tableView dequeueReusableCellWithIdentifier:[ACDTitleDetailCell reuseIdentifier]];
    if (!cell) {
        cell = [[ACDTitleDetailCell alloc] initWithStyle:UITableViewCellStyleValue1 reuseIdentifier:[ACDTitleDetailCell reuseIdentifier]];
    }
    if (indexPath.row == 0) {
        
        cell.nameLabel.attributedText = [self titleAttribute:@"UI Library Version"];
        NSString *ver = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleShortVersionString"];
        
        NSString *detailContent = [NSString stringWithFormat:@"AgoraChat v:%@",ver];
        cell.detailLabel.attributedText = [self detailAttribute:detailContent];
        
    } else if (indexPath.row == 1) {
        
        cell.nameLabel.attributedText = [self titleAttribute:@"SDK Version"];
        NSString *detailContent = [NSString stringWithFormat:@"AgoraChat v:%@",[[AgoraChatClient sharedClient] version]];
        cell.detailLabel.attributedText = [self detailAttribute:detailContent];
    }else if (indexPath.row == 2) {
        
        cell.nameLabel.attributedText = [self titleAttribute:@"More"];
        
        NSAttributedString *attributeString = [ACDUtil attributeContent:@"Agora.io" color:TextLabelBlueColor font:Font(@"PingFang SC",16.0)];
        cell.detailLabel.attributedText = attributeString;
        ACD_WS
        cell.tapCellBlock = ^{
            [weakSelf goAgoraOffical];
        };
    }
    
    return cell;
}


- (void)goAgoraOffical {
    NSString *urlString = @"https://www.agora.io/en";
    ACDWebViewController *webVC = [[ACDWebViewController alloc] initWithURLString:urlString];
    [self.navigationController pushViewController:webVC animated:YES];
}

- (NSAttributedString *)titleAttribute:(NSString *)title {
    return [ACDUtil attributeContent:title color:TextLabelBlack2Color font:Font(@"PingFang SC",16.0)];
}

- (NSAttributedString *)detailAttribute:(NSString *)detail {
    return [ACDUtil attributeContent:detail color:TextLabelGrayColor font:Font(@"PingFang SC",16.0)];
}

 
@end

File: AgoraChat-Swift/AgoraChat-Swift/code/Controllers/Settings/AgoraAboutViewController.swift
//
//  AgoraAboutViewController.swift
//  AgoraChat-Swift
//
//  Created by 冯钊 on 2022/10/12.
//

import UIKit

class AgoraAboutViewController: UITableViewController {

    @IBOutlet weak var versionLabel: UILabel!
    
    @IBOutlet weak var sdkVersionLabel: UILabel!
    override func viewDidLoad() {
        super.viewDidLoad()

        let backButton = UIButton(frame: CGRect(x: 0, y: 0, width: 8, height: 15))
        backButton.setImage(UIImage(named: "black_goBack"), for: .normal)
        backButton.setTitle("About".localized, for: .normal)
        backButton.setTitleColor(.black, for: .normal)
        backButton.addTarget(self, action: #selector(back), for: .touchUpInside)
        self.navigationItem.leftBarButtonItem = UIBarButtonItem(customView: backButton)
        
        let ver = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
        self.versionLabel.text = "AgoraChat v:\(ver)"
        
        let sdkVer = "AgoraChat v:\(AgoraChatClient.shared().version)"
        self.sdkVersionLabel.text = sdkVer
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        self.navigationController?.setNavigationBarHidden(false, animated: true)
    }
    
    @objc private func back() {
        self.navigationController?.popViewController(animated: true)
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        if indexPath.row == 2 {
            if let webVC = ACDWebViewController(urlString: "https://www.agora.io/en") {
                self.navigationController?.pushViewController(webVC, animated: true)
            }
        }
    }
    
    override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
        return 0.1
    }
}
